RastenbergerAngularJS/src/main/webapp/foodoffers.js

```diff
@@ -0,0 +1,54 @@
+var app = angular.module("foodoffer-directives", []);
+
+app.directive("foodofferList", function () {
+    return {
+        restrict: "E",
+        templateUrl: "foodoffer-list.html",
+        controller: ["$http", "$log", function ($http, $log) {
+                $log.log("Getting all foodoffers from REST resource.");
+
+                var controller = this;
+                controller.foodoffers = [];
+
+                $http({
+                    method: "GET",
+                    url: "http://localhost:8080/RastenbergerRest/rs/foodoffers"
+                }).then(function successCallback(response) {
+                    controller.foodoffers = response.data;
+                }, function errorCallback(response) {
+                    controller.foodoffers = [];
+                });
+            }
+        ],
+        controllerAs: "foodofferCtrl"
+    };
+});
+
+app.directive("foodofferForm", function () {
+    return {
+        restrict: "E",
+        templateUrl: "foodoffer-form.html",
+        controller: ["$http", "$log", function ($http, $log) {
+                var controller = this;
+                controller.foodoffer = {};
+
+                controller.submitFoodoffer = function () {
```

Comment:
This code successfully creates a foodoffer. Unfortunately there are still two remaining problems.

1. The urls for the REST services are hard coded. (Same like method above)
2. The page is not refreshed after the update. (I'm open for suggestions how this should be done)